apis/v1alpha2/validation/httproute_test.go

```diff
@@ -431,3 +431,119 @@ func TestValidateHTTPPathMatch(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateServicePort(t *testing.T) {
+	portPtr := func(n int) *gatewayv1a2.PortNumber {
+		p := gatewayv1a2.PortNumber(n)
+		return &p
+	}
+
+	groupPtr := func(g string) *gatewayv1a2.Group {
+		p := gatewayv1a2.Group(g)
+		return &p
+	}
+
+	kindPtr := func(k string) *gatewayv1a2.Kind {
+		p := gatewayv1a2.Kind(k)
+		return &p
+	}
+
+	tests := []struct {
+		name     string
+		route    *gatewayv1a2.HTTPRoute
+		errCount int
+	}{
+		{
+			name:     "default groupkind with port",
+			errCount: 0,
+			route: &gatewayv1a2.HTTPRoute{
+				Spec: gatewayv1a2.HTTPRouteSpec{
+					Rules: []gatewayv1a2.HTTPRouteRule{{
+						BackendRefs: []gatewayv1a2.HTTPBackendRef{{
+							BackendRef: gatewayv1a2.BackendRef{
+								BackendObjectReference: gatewayv1a2.BackendObjectReference{
+									Port: portPtr(99),
+								},
+							},
+						}},
+					}},
+				},
+			},
+		}, {
+			name:     "default groupkind with no port",
+			errCount: 1,
+			route: &gatewayv1a2.HTTPRoute{
+				Spec: gatewayv1a2.HTTPRouteSpec{
+					Rules: []gatewayv1a2.HTTPRouteRule{{
+						BackendRefs: []gatewayv1a2.HTTPBackendRef{{
+							BackendRef: gatewayv1a2.BackendRef{
+								BackendObjectReference: gatewayv1a2.BackendObjectReference{},
```

Comment:
Can we at least add a Name to the BackendObjectReference object everywhere? Minor but helps with readability.